### PR TITLE
{AKS} `az aks update`: Add `--network-plugin` to support updating cluster networking settings

### PR DESCRIPTION
**Pull request by tyler-lloyd**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az aks update --network-plugin azure`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ To release a new version, please select a new version number (usually plus 1 to
 Pending
 +++++++
 * Vendor new SDK and bump API version to 2023-09-02-preview.
+* Add `--network-plugin` to the `az aks update` command.
```
Comment (tyler-lloyd) [inline src/aks-preview/HISTORY.rst:15]: @zhoxing-ms , @FumingZhang - should I just bump this to 0.5.168 or does that happen elsewhere?

**Comment by tyler-lloyd** [comment]
this does not include a live test because the toggle is not enabled right now. I will come back and add a live test when we flip the toggle, next week?

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**